src/iris/pipelines/iris_pipeline.py

```diff
@@ -74,244 +81,89 @@ def __init__(
-                            )
-                elif isinstance(param_value, PipelineClass):
-                    current_node.algorithm.params[param_name] = instantiate_class_from_name(
-                        class_name=param_value.class_name, kwargs=param_value.params
-                    )
-            instanciated_pipeline.append(current_node)
-        return instanciated_pipeline
-
-    def instanciate_node(
-        self, node_class: str, algorithm_params: Dict[str, Any], callbacks: Optional[List[PipelineClass]]
-    ) -> Algorithm:
-        """Instanciate an Algorithm from its class, kwargs and optional Callbacks.
-
-        NOTE: All callbacks of type listed in self.env.disabled_qa will be filtered out. This allows one config file to be used in various QA standards levels.
-
+        Custom error handling for node execution in IRISPipeline.
+        Handles PipelineCallTraceStorageError and KeyError by checking if the node's class is in disabled_qa.
         Args:
-            node_class (str): Node's class.
-            algorithm_params (Dict[str, Any]): Node's kwargs.
-            callbacks (Optional[List[PipelineClass]]): list of callbacks.
-
+            node (PipelineNode): The node where the error occurred.
+            error (Exception): The exception raised during node execution.
         Returns:
-            Algorithm: instanciated node.
+            bool: True if the error was skipped, False otherwise.
         """
-        if callbacks is not None and len(callbacks):
-            instanciated_callbacks = [instantiate_class_from_name(cb.class_name, cb.params) for cb in callbacks]
-            instanciated_callbacks = [cb for cb in instanciated_callbacks if type(cb) not in self.env.disabled_qa]
-
-            algorithm_params = {**algorithm_params, **{"callbacks": instanciated_callbacks}}
-
-        return instantiate_class_from_name(node_class, algorithm_params)
-
-    def _check_pipeline_coherency(self, params: Parameters) -> None:
-        """Check the pipeline configuration coherency.
-
-        Args:
-            params (Parameters): Pipeline parameters.
+        # If the error is a PipelineCallTraceStorageError or KeyError and the node is in disabled_qa, skip
+        if isinstance(error, (PipelineCallTraceStorageError, KeyError)):
+            node_class = pydoc.locate(node.algorithm.class_name)
+            if node_class in getattr(self.env, "disabled_qa", []):
+                return True
+            # Otherwise, treat as a missing node error
+            self.env.error_manager(self.call_trace, ValueError(f"Could not find node {node.name}."))
+            return False
+        else:
+            # For all other errors, use the default error manager
+            self.env.error_manager(self.call_trace, error)
+            return False
 
-        Raises:
-            IRISPipelineError: Raised if a node's inputs are not declared beforehands
+    @classmethod
```

Comment:
@wiktorlazarski this is now a `@classmethod` in order to be able to properly propagate class hierarchy, i.e. priv_iris does not need to overwrite this method anymore afaik